### PR TITLE
Making Collection::sumOf() use an identity by default

### PR DESCRIPTION
**Pull request by lorenzo**
During the workshop I realized that sumOf() should not require a first parameter every time

### CODE REVIEW

**Comment by dereuromark** [comment]
:+1:

File: src/Collection/CollectionTrait.php [hunk 1]
```diff
@@ -242,8 +242,12 @@ public function countBy($callback)
      * {@inheritDoc}
      *
      */
-    public function sumOf($matcher)
+    public function sumOf($matcher = null)
     {
+        if ($matcher === null) {
+            return array_sum($this->toList());
```
Comment (markstory) [inline src/Collection/CollectionTrait.php]: very optimize.

File: src/Collection/CollectionInterface.php [hunk 2]
```diff
@@ -444,13 +444,17 @@ public function countBy($callback);
...
+     * // Total: 6
      * ```
      *
      * @param string|callable $matcher The property name to sum or a function
+     * If no value is passed, an identity function will be used.
      * that will return the value of the property to sum.
      * @return float|int
      */
-    public function sumOf($matcher);
+    public function sumOf($matcher = null);
```
Comment (markstory) [inline src/Collection/CollectionInterface.php]: This _could_ be a backwards compatible break. But I don't think there will be userland implementations of CollectionInterface.

Comment (lorenzo) [inline src/Collection/CollectionInterface.php]: well adding methods to the interface is more breaking than changing the default. I doubt that anyone is implementing the full interface as you said

**Pull request merged**